### PR TITLE
Ignore fixed slots in ad-top-of-viewport audit

### PR DESCRIPTION
**Pull request by jburger424**

### CODE REVIEW

File: lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js [hunk 1]
```diff
@@ -78,7 +79,7 @@ class AdTopOfViewport extends Audit {
       return auditNotApplicable(NOT_APPLICABLE.NO_ADS_VIEWPORT);
     }
 
-    const score = inViewport && topSlot.midpoint < 200 ? 0 : 1;
+    const score = inViewport && topSlot.midpoint < 100 ? 0 : 1;
```
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js]: Nit: move 100 to a (possibly local) constant with semantically meaningful name

File: lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js [hunk 2]
```diff
@@ -78,7 +79,7 @@ class AdTopOfViewport extends Audit {
...
 
     return {
       score,
```
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js]: How about having a non binary score here? Something like:
```
const score = Math.min(topSlot.midpoint, 100) / 100
```

But maybe the score should be non-linear. Something like `Math.min((topSlot.midpoint / 100) ** 2, 1)`

File: lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js [hunk 3]
```diff
@@ -19,6 +19,43 @@ const pageFunctions = require('lighthouse/lighthouse-core/lib/page-functions.js'
...
+ */
+function isFixed(element) {
+  // @ts-ignore
+  const scrollableAncestor = ((element) => {
+    let currentEl = element.parentElement;
+    while (currentEl) {
+      if (currentEl.scrollHeight > currentEl.clientHeight) {
+        // Check style before computedStyle as computedStyle is expensive.
+        const yScroll = currentEl.style.overflowY ||
+          window.getComputedStyle(currentEl).overflowY;
```
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js]: How about moving `el.style.foo || window.getComputedStyle(currentEl).foo` to a helper function?

File: lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js [hunk 4]
```diff
@@ -19,6 +19,43 @@ const pageFunctions = require('lighthouse/lighthouse-core/lib/page-functions.js'
...
 /* eslint-env browser, node */
 
+/**
+ * TODO(jburger): Move to page-functions.js once integrated with Lighthouse.
+ * @param {?HTMLElement} element
+ * @return {Boolean}
+ */
+function isFixed(element) {
+  // @ts-ignore
+  const scrollableAncestor = ((element) => {
```
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js]: `hasScrollableAncestor`? Prefer function names with verbs (except for handlers)

File: lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js [hunk 5]
```diff
@@ -19,6 +19,43 @@ const pageFunctions = require('lighthouse/lighthouse-core/lib/page-functions.js'
...
+        const yScroll = currentEl.style.overflowY ||
+          window.getComputedStyle(currentEl).overflowY;
+        if (yScroll) {
+          return true;
+        }
+      }
+      currentEl = currentEl.parentElement;
+    }
+    return false;
+  });
```
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js]: Nit: remove extra parens

File: lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js (hunk 4 above)
Comment (warrengm) [inline lighthouse-plugin-ad-speed-insights/gatherers/iframe-elements.js]: Is it worth pulling this helper outside of this function? It doesn't look like it closes over anything. In fact it shadows the outer `element` which isn't great

File: lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js (hunk 2 above)
Comment (jburger424) [reply lighthouse-plugin-ad-speed-insights/audits/ad-top-of-viewport.js]: I like the idea but will defer to a later PR so I am able to thoroughly test against multiple pubs.

**Pull request merged**